homeassistant/components/nexia/scene.py

```diff
@@ -36,33 +32,25 @@ class NexiaAutomationScene(NexiaEntity, Scene):
 
     def __init__(self, coordinator, automation):
         """Initialize the automation scene."""
-        super().__init__(coordinator)
+        super().__init__(
+            coordinator, name=automation.name, unique_id=automation.automation_id,
+        )
         self._automation = automation
 
-    @property
-    def unique_id(self):
-        """Return the unique id of the automation scene."""
-        # This is the automation unique_id
-        return self._automation.automation_id
-
-    @property
-    def name(self):
-        """Return the name of the automation scene."""
-        return self._automation.name
-
     @property
     def device_state_attributes(self):
         """Return the scene specific state attributes."""
-        return {
-            ATTR_ATTRIBUTION: ATTRIBUTION,
-            ATTR_DESCRIPTION: self._automation.description,
-        }
+        data = super().device_state_attributes
+        data.update({ATTR_DESCRIPTION: self._automation.description})
+        return data
 
     @property
     def icon(self):
         """Return the icon of the automation scene."""
         return "mdi:script-text-outline"
 
-    def activate(self):
+    async def async_activate(self):
         """Activate an automation scene."""
-        self._automation.activate()
+        await self.hass.async_add_executor_job(self._automation.activate)
+        await asyncio.sleep(SCENE_ACTIVATION_TIME)
```

Comment:
Shouldn't we just schedule an async refresh to be done after `SCENE_ACTIVATION_TIME` ? The command has already been issued, it's just the state update causing us to sleep. That means that any script/automation will now also have to wait for the state change, that seems unnecesssary.

Reply:
Yes, sleep is the solution of last resort :)

Reply:
switched to async_call_later

Reply:
That didn't work
```
Exception in callback async_track_point_in_utc_time.<locals>.point_in_time_listener(<Event time_c....003775-05:00>) at /usr/src/homeassistant/homeassistant/helpers/event.py:234
```
I screwed something up.. rechecking

Reply:
fixed